src/main/java/sg/ncl/MainController.java

```diff
@@ -254,6 +268,41 @@ public String testbedInformation(Model model) throws IOException {
         return "testbedInformation";
     }
 
+    // get all the nodes' status
+    // there are three types of status
+    // "free" : node is free
+    // "in_use" : node is in use
+    // "reload" : node is in process of freeing or unknown status
+    // "reserved" : node is pre-reserved for a project
+    @RequestMapping("/testbedNodesStatus")
+    public String testbedNodesStatus(Model model) throws IOException {
+
+        Map<MachineType, List<Map<String, String>>> nodesStatus = getNodesStatus();
+        Map<MachineType, Map<String, Long>> nodesStatusCount = new HashMap<>(); // count the number of different nodes status, e.g. SYSTEMX = { FREE = 10, IN_USE = 11, ... }
+
+        // loop through each of the machine type
+        // tabulate the different nodes type
+        nodesStatus.entrySet().forEach(machineTypeListEntry -> {
+            Map<String, Long> nodesCountMap = new HashMap<>();
+
+            long free = machineTypeListEntry.getValue().stream().filter(stringStringMap -> stringStringMap.get("status").equalsIgnoreCase("free")).count();
+            long in_use = machineTypeListEntry.getValue().stream().filter(stringStringMap -> stringStringMap.get("status").equalsIgnoreCase("in_use")).count();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move the "in_use" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00117)